libsolidity/formal/CHC.h

```diff
@@ -275,12 +270,24 @@ class CHC: public SMTEncoder
 	//@{
 	struct CHCVerificationTarget: VerificationTarget
 	{
-		smtutil::Expression errorId;
+		unsigned errorId;
+		ASTNode const * const errorNode;
+	};
+
+	struct CHCVerificationTargetPlaceholder
+	{
+		smtutil::Expression constraints;
+		smtutil::Expression errorExpression;
+		smtutil::Expression startingBlock;
 	};
 
 	/// Verification targets corresponding to ASTNodes. There can be multiple targets for a single ASTNode,
 	/// e.g., divByZero and Overflow for signed division.
-	std::map<ASTNode const*, std::vector<CHCVerificationTarget>, IdCompare> m_verificationTargets;
+	std::map<ASTNode const*, std::vector<unsigned>, IdCompare> m_functionVerificationPreTargetIds;
+	std::map<unsigned, CHCVerificationTarget> m_verificationPreTargets;
+
+	/// There can be multiple contexts for the same function, e.g., with contract inheritance
```

Comment:
I think we need to document these variables better. Maybe add before this comment, "Query placeholders for constructors, if the key has type ContractDefinition*, or external functions, if the key has type FunctionDefinition*."

Reply:
I have tried to add some documentation, including your suggestion.